app/assets/javascripts/disablePdfLink.js

```diff
@@ -1,20 +1,26 @@
 $( document ).on('turbolinks:load', function() {
-    checkLinkStatus();
+    if ($('.download-pdf').length) {
+        checkLinkStatus();
+    }
 });
 
 function checkLinkStatus() {
-    let pdfUrl = $('.download-pdf').attr('href');
-    $.ajax({
-        url: pdfUrl,
-        success: function() {
-            $('.download-pdf').text('A PDF version of your completed form is now available for download')
-                .css('pointer-events', 'auto');
-        },
-        error: function() {
-            $('.download-pdf').text('Not quite ready yet - a PDF version of your completed form will be available for download shortly')
-                .css('pointer-events', 'none');
-            console.warn(`Unable to find PDF, retrying ${pdfUrl} 10 seconds`);
-            setTimeout(checkLinkStatus, 10000);
+    let url = $('.download-pdf').attr('href');
+    let http = new XMLHttpRequest();
+    http.overrideMimeType("application/pdf");
+    http.open('GET', url, true);
+    http.onload = function (e) {
+        if (http.readyState === 4) {
+            if (http.status === 200) {
+                $('.download-pdf').text('A PDF version of your completed form is now available for download')
+                    .css('pointer-events', 'auto');
+            } else {
+                $('.download-pdf').text('Not quite ready yet - a PDF version of your completed form will be available for download shortly')
+                    .css('pointer-events', 'none');
+                console.warn(`Unable to find PDF, retrying ${url} in 10 seconds`);
+                setTimeout(function() { checkLinkStatus() }, 10000);
```

Comment:
Does this not mean that on every call to checkLinkStatus when the file doesnt exist, a new 'setTimeout' is called - meaning lots and lots and lots of checkLinkStatus methods will start to get called as time goes on ?
Normally, you set it once and cancel it when done ?

Reply:
No, I think you're thinking of `setInterval`. Please see below video demonstration, this works as expected with no additional calls.

https://streamable.com/bifkk

Reply:
Cool - youre right I was thinking of setInterval - cheers